sqlobject/src/main/java/org/jdbi/v3/sqlobject/mixins/SqlObjectMixin.java

```diff
@@ -16,20 +16,34 @@
 import org.jdbi.v3.core.Handle;
 import org.jdbi.v3.core.HandleCallback;
 import org.jdbi.v3.core.HandleConsumer;
+import org.jdbi.v3.core.HandleSupplier;
 
 /**
- * A mixin interface to define a method for obtaining the Handle attached to a sql object
+ * SqlObject base mixin interface.  Custom mixins may be written as subinterfaces
+ * of this class with default methods.
  */
-public interface GetHandle
+public interface SqlObjectMixin<This extends SqlObjectMixin<This>>
 {
     /**
-     * Obtain the handle associated to this sql object.
-     *
-     * @return the handle
+     * @return the {@link HandleSupplier} bound to this sql object.
+     */
+    HandleSupplier getHandleSupplier();
+
+    /**
+     * @return the handle open in the current sql object context.
      */
     Handle getHandle();
```

Comment:
I'm wondering if mixins even need to be a thing. e.g.

```java
@SqlMethodAnnotation(GetHandle.Factory.class)
@interface GetHandle {
  class Factory implements HandlerFactory {
    public Handler buildHandler(Class<?> sqlObjectType, Method method) {
      return (target, args, handleSupplier) -> handleSupplier.getHandle();
    }
  }
}

interface SqlObjectMixin {
  @GetHandle Handle getHandle();
}

interface Transactional<This extends Transactional<This>> extends SqlObjectMixin {
  ...
}
```

With this approach, the concept of mixins would be implied by inheritance, and `SqlObjectFactory` wouldn't need any special case code for mixins.